### DISCUSSION TITLE
Is that OK to use "rewrites" and "output: export" only for local development?

### DISCUSSION THREAD

**Discussion by ronny1020** [vercel/next.js#62972] [Help]
### Summary

Hi guys,

I have an "output: export" SSG project with the app router.

Since it's a static site, I fetch the data from the client side. The API and the site would be on the same domain when it comes to production. Therefore, I need an HTTP proxy to get the API for local development and I think the simple way to have a proxy with next.js is through the "rewrites" feature.

Then I get this warning message.
```bash
 ⚠ Specified "rewrites" will not automatically work with "output: export". See more info here: https://nextjs.org/docs/messages/export-no-custom-routes
```

However, the proxy is still working properly on the local. That confuses me is this the proper way to use an HTTP proxy locally with "output: export"? If it's OK, can I turn off the message? Or if it's NOT OK, what would be the proper way to use the proxy with "output: export"?

Thanks.

### Additional information

```js
⚠ Specified "rewrites" will not automatically work with "output: export". See more info here: https://nextjs.org/docs/messages/export-no-custom-routes

Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:10 PST 2023; root:xnu-10002.61.3~2/RELEASE_X86_64
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 1.22.19
  pnpm: 8.15.2
Relevant Packages:
  next: 14.1.1
  eslint-config-next: 14.1.1
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: export
```

**Comment by clement-escolano** [comment]
I use it in the same way so I hope this is a good practice :-)

To solve the "warning" issue when running locally, I set this in `next.config.js`:

```js
output: process.env.NODE_ENV !== "development" ? "export" : "standalone",
```
